src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -71,6 +71,8 @@ public virtual IDisposable Start()
                 async features =>
                 {
                     var httpContext = contextFactory.CreateHttpContext(features);
+                    httpContext.Response.RegisterForDispose(httpContext);
```

Comment:
I think its the weirdest code I've written...

Most sensibly it should return Task<IDisposable> rather than Task; but then' I'd have to add a PR to Kestrel as well... Ok will do that?